### ISSUE TITLE
Import-AzApiMangementApi docs incorrect

### ISSUE THREAD

**Issue by dpbevin** [Azure/azure-powershell#8728]
## Description

When attempting to **update** an API using `Import-AzApiManagementApi` I found it was required to specify the `ApiId` parameter. The documentation states that ApiId is optional parameter.

The docs followed were [https://docs.microsoft.com/en-us/powershell/module/azurerm.apimanagement/import-azurermapimanagementapi?view=azurermps-6.13.0](docs)

## Steps to reproduce

```powershell
# Create a blank API
$api = New-AzApiMangementApi -Context $ctx -ApiId "blank-import" -Name "Blank Then Import" -ServiceUrl "https://foo.woo.com" -Protocols @("https") -Path "blankimport"

# Now attempt to import without specifying ApiId
Import-AzApiManagementApi -Context $ctx -SpecificationFormat Swagger -SpecificationPath "petstore-swagger" -Path "blankimport"

# Repeating the above step but adding -ApiId "blank-import" worked
```

## Environment data

```
Name                           Value
----                           -----
PSVersion                      5.1.17134.407
PSEdition                      Desktop
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
BuildVersion                   10.0.17134.407
CLRVersion                     4.0.30319.42000
WSManStackVersion              3.0
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
```

## Module versions

```powershell
ModuleType Version    Name                                ExportedCommands
---------- -------    ----                                ----------------
Script     1.3.1      Az.Accounts                         {Disable-AzDataCollection, Disable-AzContextAutosave, Enable-AzDataCollection, Enable-AzContextAutosave...}
Script     1.0.1      Az.Aks                              {Get-AzAks, New-AzAks, Remove-AzAks, Import-AzAksCredential...}
Script     1.0.2      Az.AnalysisServices                 {Resume-AzAnalysisServicesServer, Suspend-AzAnalysisServicesServer, Get-AzAnalysisServicesServer, Remove-AzAnalysisServicesServer...}
Script     1.0.0      Az.ApiManagement                    {Add-AzApiManagementRegion, Get-AzApiManagementSsoToken, New-AzApiManagementCustomHostnameConfiguration, New-AzApiManagementSystemCertificate...}
```

## Debug output

```
DEBUG: 4:37:39 PM - ImportAzureApiManagementApi begin processing with ParameterSet 'ImportFromLocalFile'.
DEBUG: 4:37:39 PM - using account id 'REDACTED'...
DEBUG: [Common.Authentication]: Authenticating using Account: 'REDACTED', environment: 'AzureCloud', tenant: 'REDACTED'
DEBUG: ============================ HTTP REQUEST ============================

HTTP Method:
PUT

Absolute Uri:
https://management.azure.com/subscriptions/REDACTED/resourceGroups/REDACTED/providers/Microsoft.ApiManagement/service/REDACTED/apis/REDACTED?api-version=2018-01-01

Headers:
x-ms-client-request-id        : REDACTED
accept-language               : en-US

Body:
{
  "properties": {
    "path": "blankimport",
    "contentValue": "{\n  \"swagger\": \"2.0\",\n  \"info\": {\n    \"description\": \"This is a sample server Petstore server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net,
#swagger](http://swagger.io/irc/).  For this sample, you can use the api key `special-key` to test the 

REMOVED SWAGGER CONTENT FOR CLARITY

  }\n    },\n    \"ApiResponse\": {\n      \"type\": \"object\",\n      \"properties\": {\n        \"code\": {\n          \"type\": \"integer\",\n          \"format\": \"int32\"\n        },\n        \"type\": {\n          \"type\": \"string\"\n
},\n        \"message\": {\n          \"type\": \"string\"\n        }\n      }\n    }\n  },\n  \"externalDocs\": {\n    \"description\": \"Find out more about Swagger\",\n    \"url\": \"http://swagger.io\"\n  }\n}",
    "contentFormat": "swagger-json"
  }
}

DEBUG: ============================ HTTP RESPONSE ============================

Status Code:
BadRequest

Headers:
Pragma                        : no-cache
Strict-Transport-Security     : max-age=31536000; includeSubDomains
x-ms-request-id               : 0402ac36-ea6e-474e-849c-38b9d28841b8
x-ms-ratelimit-remaining-subscription-writes: 1198
x-ms-correlation-request-id   : 52a652e1-5189-43d6-9245-05c166dd8ea5
x-ms-routing-request-id       : AUSTRALIASOUTHEAST:20190308T053742Z:52a652e1-5189-43d6-9245-05c166dd8ea5
X-Content-Type-Options        : nosniff
Cache-Control                 : no-cache
Date                          : Fri, 08 Mar 2019 05:37:41 GMT
Server                        : Microsoft-HTTPAPI/2.0

Body:
{
  "error": {
    "code": "ValidationError",
    "message": "One or more fields contain incorrect values:",
    "details": [
      {
        "code": "ValidationError",
        "target": "name",
        "message": "API with specified name already exists"
      }
    ]
  }
}

Import-AzApiManagementApi : Operation returned an invalid status code 'BadRequest'
At line:1 char:16
+ ... iBlankPet = Import-AzApiManagementApi -Context $apimContext -Specific ...
+                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Import-AzApiManagementApi], ErrorResponseException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands.ImportAzureApiManagementApi

DEBUG: AzureQoSEvent: CommandName - Import-AzApiManagementApi; IsSuccess - False; Duration - 00:00:02.3559283; Exception - Microsoft.Azure.Management.ApiManagement.Models.ErrorResponseException: Operation returned an invalid status code 'BadRequest'
   at Microsoft.Azure.Management.ApiManagement.ApiOperations.<CreateOrUpdateWithHttpMessagesAsync>d__8.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.ApiManagement.ApiOperationsExtensions.<CreateOrUpdateAsync>d__7.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.Azure.Management.ApiManagement.ApiOperationsExtensions.CreateOrUpdate(IApiOperations operations, String resourceGroupName, String serviceName, String apiId, ApiCreateOrUpdateParameter parameters, String ifMatch)
   at Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands.ImportAzureApiManagementApi.ExecuteApiManagementCmdlet()
   at Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands.AzureApiManagementCmdletBase.ExecuteCmdlet();
DEBUG: Finish sending metric.
```

## Error output

```
Message        : Operation returned an invalid status code 'BadRequest'
StackTrace     :    at Microsoft.Azure.Management.ApiManagement.ApiOperations.<CreateOrUpdateWithHttpMessagesAsync>d__8.MoveNext()
                 --- End of stack trace from previous location where exception was thrown ---
                    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                    at Microsoft.Azure.Management.ApiManagement.ApiOperationsExtensions.<CreateOrUpdateAsync>d__7.MoveNext()
                 --- End of stack trace from previous location where exception was thrown ---
                    at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
                    at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
                    at Microsoft.Azure.Management.ApiManagement.ApiOperationsExtensions.CreateOrUpdate(IApiOperations operations, String resourceGroupName, String serviceName, String apiId, ApiCreateOrUpdateParameter parameters, String ifMatch)
                    at Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands.ImportAzureApiManagementApi.ExecuteApiManagementCmdlet()
                    at Microsoft.Azure.Commands.ApiManagement.ServiceManagement.Commands.AzureApiManagementCmdletBase.ExecuteCmdlet()
Exception      : Microsoft.Azure.Management.ApiManagement.Models.ErrorResponseException
InvocationInfo : {Import-AzApiManagementApi}
Line           : $api = Import-AzApiManagementApi -Context $ctx -SpecificationFormat Swagger -SpecificationPath "C:\temp\petstore-swagger.json" -Path "blankimport"
Position       : At line:1 char:16
                 + ... apii = Import-AzApiManagementApi -Context $apimContext -Specific ...
                 +                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by dpbevin** [comment]
Similar to https://github.com/Azure/azure-powershell/issues/6965

**Comment by cormacpayne** [comment]
@solankisamir Hey Samir, would you mind taking a look at this issue?

**Comment by solankisamir** [comment]
@dpbevin the `Import-AzApiManagementApi` is supposed to work in 2 cases
1. When you just want to import an Openapi spec into a brand new API (don't care of the ApiId)
2. When you want to import an Openapi spec into an existing API

The common denominator for these 2 cases is `-ApiId` parameter is `optional`. 
When importing an API `name` parameter is derived from the OpenApi spec. The `name` in an Api Management service instance for Api have to be unique, that is the reason why your example throws the error.

**Comment by solankisamir** [comment]
Provided input, please open if you feel the information is incorrect

**Comment by beatrichartz** [comment]
@solankisamir We just ran into this - it's quite opaque that `Import-AzApiManagementApi` does not respect the input for `ApiId` and prefers the title attribute of a Swagger specification. In our case, because we needed to make the same specification available in two APIs, this lead to a situation where `Import-AzApiManagementApi` tried to import an API with the same specification under the same API title - the first import works fine, but subsequent imports fail with validation errors / bad request. The workaround now is that we download the swagger spec, replace the title, and then use the import.

I'd suggest to use `ApiId` as the parameter to upsert against for usecase 2 (When you want to import an Openapi spec into an existing API) if it is present.
